src/rust/ide/view/graph-editor/src/selection.rs

```diff
@@ -0,0 +1,422 @@
+//! Module that contains the logic sor selecting nodes. This includes selecting single nodes
+//! by clicking on them separately, as well as click+drag for selecting with a selection area.
+mod bounding_box;
+
+use ensogl::prelude::*;
+
+pub use bounding_box::BoundingBox;
+
+use crate::NodeId;
+use crate::Nodes;
+use crate::TouchState;
+
+use ensogl::frp;
+use ensogl::gui::cursor;
+use ensogl::gui::cursor::Cursor;
+
+
+
+// ============
+// === Mode ===
+// ============
+
+/// Possible selection modes.
+#[derive(Clone,Copy,Debug,Eq,PartialEq)]
+pub enum Mode {
+    /// Select a single node when clicking on it. Deselects all other nodes.
+    Normal,
+    /// Toggle the selection state of the select node without changing the selection state of
+    ///other node. This allows to add and remove nodes from the current selection.
+    Multi,
+    /// Add selected nodes to the set of currently selected nodes.
+    Merge,
+    /// Remove selected nodes from the set of currently selected nodes.
+    Subtract,
+    /// Invert the selection state of the selected nodes.
+    Inverse
+}
+
+impl Mode {
+
+    /// Return whether an element should be selected, if a selection was triggered through single
+    /// selection and had the given `was_selected` status before.
```

Comment:
Ok, so here "single selection" is an opposite of "area selection" - but I think this is understandable only in context of the other methods. Can we add a short explanation here, and in the method below?
```suggestion
    /// Return whether an element should be selected, if a selection was triggered through single
    /// (not area) selection and had the given `was_selected` status before.
```